### DISCUSSION TITLE
Return specific error messages during `/stream` and `/astream_events`

### DISCUSSION THREAD

**Discussion by philbo** [langchain-ai/langserve#545] [Q&A] [answered]
I would like to be able to add specific error messages to the streamed responses so my client can respond appropriately.

Currently it seems that any exception is purposefully [hardcoded](https://github.com/langchain-ai/langserve/blob/e1cec832da29173df89bfe50b35be8ce59b1a8b5/langserve/api_handler.py#L984-L986) to return:
```
"event": "error",
"data": json.dumps(
      {"status_code": 500, "message": "Internal Server Error"}
  ),
```

Is there any way to change this behaviour?

To give some wider context, say I have a Chain which scrapes a URL provided by the client and then passes that response content to the LLM with some prompt:
```
fetch_url | prompt | llm | parser
```
If the URL is invalid, or returns an error response, I want to stop the chain and return a useful message to the client. I can currently do this with the `/invoke` endpoint by raising different types of `HTTPException` errors in my chain.

**Comment by eyurtsev** [comment, marked as answer]
Here's an issue for this: https://github.com/langchain-ai/langserve/issues/481

Duplicate of https://github.com/langchain-ai/langserve/discussions/473
